pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/PrometheusMetricStreams.java

```diff
@@ -65,11 +65,19 @@ void releaseAll() {
         metricStreamMap.clear();
     }
 
-    private SimpleTextOutputStream initGaugeType(String metricName) {
+    private SimpleTextOutputStream initMetricType(String metricName) {
+        String metricTypeDef = String.format("# TYPE %s %s\n", metricName, metricType(metricName));
```

Comment:
Nit: this might have unintended consequences for performance. With this change, we're creating the string for every call to this method instead of only when the metric has not yet been initialized. I'm not sure how much it matters in this section, but I know the prometheus metrics generation has been optimized because it was generating unnecessary load.